department-of-veterans-affairs/caseflow
# Justin/APPEALS-49739

Resolves [Jira Issue Title](https://jira.devops.va.gov/browse/JIRA-12345)

# Description
Created a modal to assign/reassign tasks for correspondence. The modal contains a user dropdown that gets dynamically filled with a reasign_users list based on whether a task is assigned to a user or organization, a task instructions text field, and a cancel and submit (Assign task) button. 

## Acceptance Criteria

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

1. Log into Caseflow as a supervisor user
2. Go to "Your Queue"
3. Click "Correspondence Cases" in the "Switch Views" drop down
4. Click any veteran in the "Veteran Details" column
5. Select any Correspondence Type and click Save Changes
6. Scroll to the bottom and click "Create Record"
7. Click Continue to go to the second step
8. Under the "Tasks not related to an Appeal" Click "+Add Task"
9. In the New Task modal that renders add and "Other Motion" task and add any test text in the instructions
10. Scroll to the bottom and click "Continue"
11. Scroll to the bottom and click "Submit" then "Confirm" in the pop-up modal
12. Copy the name that shows up in the support banner on the next redirected page(/correspondence/team)
13. Go to the "Pending" tab
14. Paste the name that you copied from the success banner
15. Click the result with "Other motion" under the "Tasks" section and copy the URL 
16. Use a new tab to log into Caseflow as a LIT SUPPORT USER
17. Return to the previous tab and refresh the page (or paste in the copied URL)
18. Scroll to the bottom click the actions dropdown in the "Tasks not related to an appeal" section
19. 


# Frontend
## User Facing Changes

> Modal to Assign task
![image](https://github.com/user-attachments/assets/d12343f6-2274-48e9-90c4-373b439cf153)

> Banner for assignment completion
![image](https://github.com/user-attachments/assets/a11652ff-7d8b-4b67-92c6-273440c77dcc)

> Dropdown to Re-assign task
![image](https://github.com/user-attachments/assets/664c2387-4d6f-4a5d-8a84-a6260373e0ac)

> Modal to Re-assign to person. Button is only clickable when user and instructions are filled out.
![image](https://github.com/user-attachments/assets/441a18d1-0400-4b24-b3b0-58bc14cb0ab0)


 BEFORE|AFTER
 ---|---
